### PR TITLE
Typos: `test/SILOptimizer`

### PR DESCRIPTION
**Pull request by Sajjon**
Fix typos in `test/SILOptimizer`

This is one batch of many PRs fixing typos, [see the tracking issue.](https://github.com/swiftlang/swift/issues/75022)

### CODE REVIEW

File: test/SILOptimizer/access_enforcement_selection.sil [hunk 1]
```diff
@@ -67,7 +67,7 @@ sil [ossa] @takesInoutAndClosure : $@convention(thin) (@inout Builtin.Int64, @gu
 sil [ossa] @closureCapturingByStorageAddress : $@convention(thin) (@inout_aliasable Builtin.Int64) -> ()
 
 // Test static enforcement of box addresses that escape via closure
-// partial_applys.
+// partial_applies.
```
Comment (eeckstein) [inline test/SILOptimizer/access_enforcement_selection.sil:70]: -> `partial_apply` instructions

File: test/SILOptimizer/access_enforcement_selection.sil [hunk 2]
```diff
@@ -99,7 +99,7 @@ bb0:
   return %9 : $()
 }
 
-sil [dynamically_replacable] [ossa] @closureCapturingByStorageAddress2 : $@convention(thin) (@inout_aliasable Builtin.Int64) -> ()
+sil [dynamically_replaceable] [ossa] @closureCapturingByStorageAddress2 : $@convention(thin) (@inout_aliasable Builtin.Int64) -> ()
```
Comment (eeckstein) [inline test/SILOptimizer/access_enforcement_selection.sil:102]: That needs to be fixed in SILPrinter + SILParser first

File: test/SILOptimizer/dictionary_lookup_with_default.swift [hunk 3]
```diff
@@ -9,7 +9,7 @@ public struct S {
     mutating func doSomething() { }
 }
 
-// Check that all partial_applys can be optimized away so that no closure context needs to be allocated.
+// Check that all partial_applies can be optimized away so that no closure context needs to be allocated.
```
Comment (eeckstein) [inline test/SILOptimizer/dictionary_lookup_with_default.swift:12]: -> `partial_apply` instructions

File: test/SILOptimizer/exclusivity_static_diagnostics.sil [hunk 4]
```diff
@@ -1479,7 +1479,7 @@ bb0(%0 : $*TestDynamic):
 
 // This callee actually accesses different subpaths, but the caller's
 // diagnostic should not be able to see that.
-sil private [dynamically_replacable] [ossa] @testDynamicLocal : $@convention(thin) (@inout Builtin.Int64, @inout_aliasable TestDynamic) -> () {
+sil private [dynamically_replaceable] [ossa] @testDynamicLocal : $@convention(thin) (@inout Builtin.Int64, @inout_aliasable TestDynamic) -> () {
```
Comment (eeckstein) [inline test/SILOptimizer/exclusivity_static_diagnostics.sil:1482]: same here

File: test/SILOptimizer/function_uses.sil [hunk 5]
```diff
@@ -27,7 +27,7 @@ sil @public_func : $@convention(thin) () -> ()
 // CHECK-NEXT:    %1 = dynamic_function_ref @internal_func
 // CHECK-NEXT:  ]
 // CHECK-NEXT:  End function internal_func
-sil hidden [dynamically_replacable] @internal_func : $@convention(thin) () -> () {
+sil hidden [dynamically_replaceable] @internal_func : $@convention(thin) () -> () {
```
Comment (eeckstein) [inline test/SILOptimizer/function_uses.sil:30]: same here

File: test/SILOptimizer/simplify_cfg_ossa_jump_threading.sil [hunk 6]
```diff
@@ -183,7 +183,7 @@ bb9:
 // not know how to allocate for multiple partial applies.
 //
 // rdar://119768691 (OwnershipModelEliminator triggers assertion when
-// lowering certain [on_stack] partial_applys in certain
+// lowering certain [on_stack] partial_applies in certain
```
Comment (eeckstein) [inline test/SILOptimizer/simplify_cfg_ossa_jump_threading.sil:186]: -> `partial_apply` instructions

**Review by eeckstein — CHANGES_REQUESTED** [review]

File: test/SILOptimizer/access_enforcement_selection.sil (hunk 2 above)
Comment (eeckstein) [reply test/SILOptimizer/access_enforcement_selection.sil:102]: I think the best is if you create a separate PR just fixing dynamically_replacable.

**Comment by Sajjon** [comment]
@eeckstein Thank you for review! I've reverted according to your suggestions.

**Review by eeckstein — APPROVED** [review]

**Comment by eeckstein** [comment]
@swift-ci smoke test

**Pull request merged**